Admin/Client/src/App.tsx

```diff
@@ -1,7 +1,57 @@
-import React from 'react';
-
-function App() {
-    return <div className='App'></div>;
-}
-
-export default App;
+import React, { useState } from 'react';
+import InitSideEffects from './sideEffects/initSideEffects';
+import LoginPage from './pages/loginPage';
+import Authentication from './authentication';
+
+import {
+    BrowserRouter as Router,
+    Switch,
+    Route,
+    Link,

+    Redirect
+} from "react-router-dom";
+import MainPage from './pages/mainPage';
+
+//Change this to move from real to mock side effects
+const isMocking = true;
+InitSideEffects(isMocking);
+
+function App() {
+    const [isLoggedin, setLoginStatus] = useState(false);
+    const login = (username: string, password: string, onStart:()=>void, onFail:(error:string)=>void, onSuccess:()=>void) => {
+        onStart();
+        Authentication.AuthenticateAsync(username, password)
+            .then(token => {
+                setLoginStatus(true);
+            })
+            .catch(err => {
+                onFail(err);
+            })
+            .finally(() => {
+                onSuccess();
+            });
+    }
+
+    return <div className='App'>
+        <Router>
+            <Switch>
+                <PrivateRoute isLoggedin={isLoggedin} component={MainPage} exact path="/" />
+                <Route exact path="/login">
+                    <LoginPage isAuthenticated={isLoggedin} login={login} />
+                </Route>
+            </Switch>
+        </Router>
+    </div>;
+}
+
+const PrivateRoute = ({ component, isLoggedin, ...rest }: any) => {

```

Comment:
why not in separate file?

Reply:
Let's run fast and make some refactoring later this type of things.
Don't forget that it's one time product - and there won't be many changes in it.
So let's make it work and close this type of cleanups later.